File: SpaceInvaders/GameObject/Bomb/FallStraight.cs
using System;
using System.Diagnostics;

namespace SpaceInvaders
{
    class FallStraight : FallStrategy
    {
        public FallStraight()
        {
            this.oldPosY = 0.0f;
        }

        public override void Reset( float posY )
        {
            this.oldPosY = posY;
        }

        public override void Fall( Bomb pBomb )
        {
            Debug.Assert(pBomb != null);
            
            // Do nothing for this strategy
        }

        // Data
        private float oldPosY;
    }
}

File: quarkus-migration/space-invaders-quarkus/src/main/java/net/beeland/spaceinvaders/strategy/FallStraight.java
package net.beeland.spaceinvaders.strategy;

import net.beeland.spaceinvaders.gameobject.Bomb;

/**
 * FallStraight - Bomb falls straight down without any horizontal movement
 */
public class FallStraight implements FallStrategy {
    
    private float oldPosY;
    
    /**
     * Constructor
     */
    public FallStraight() {
        this.oldPosY = 0.0f;
    }
    
    @Override
    public void reset(float posY) {
        this.oldPosY = posY;
    }
    
    @Override
    public void fall(Bomb bomb) {
        // Do nothing - bomb falls straight down naturally
        // No horizontal movement or scale changes
    }
    
    /**
     * Get the old Y position
     * 
     * @return Old Y position
     */
    public float getOldPosY() {
        return this.oldPosY;
    }
}